src/components/Transcript.tsx

```diff
@@ -0,0 +1,347 @@
+    if (!str || !textRef.current) {
+      return;
+    }
+    const ranges: Range[] = [];
+    for (const rowElement of textRef.current.childNodes) {
+      const textNode = rowElement.lastChild?.firstChild;
+      if (!textNode) {
+        continue;
+      }
+      const content = textNode?.textContent?.toLowerCase();
+      if (!content) {
+        continue;
+      }
+      let index = content.indexOf(str);
+      while (index > -1) {
+        const range = new Range();
+        range.setStart(textNode, index);
+        range.setEnd(textNode, index + str.length);
+        ranges.push(range);
+
+        index = content.indexOf(str, index + 1);
+      }
+    }
+    console.log(ranges);
+    const highlight = new Highlight(...ranges);
+    CSS.highlights.set("search-results", highlight);
+  }, [searchTerm]);
+
+  return (
+    <div
+      css={css`
+        margin: 93px auto;
+        max-width: 860px;
+        text-align: left;
+        display: flex;
+        padding: var(--leo-spacing-7xl);
+        flex-direction: column;
+        align-items: center;
+        gap: var(--leo-spacing-4xl);
+        align-self: stretch;
+        border-radius: var(--leo-radius-xl);
+        background: var(--leo-color-container-background);
+        box-shadow:
+          0px var(--Elevation-xxs, 1px) 0px 0px
+            var(--Semantic-Elevation-Primary, rgba(0, 0, 0, 0.05)),
+          0px var(--Elevation-xxs, 1px) var(--Elevation-xs, 4px) 0px
+            var(--Semantic-Elevation-Secondary, rgba(0, 0, 0, 0.1));
+      `}
+    >
+      <div
+        css={css`
+          display: flex;
+          align-items: center;
+          align-self: stretch;
+        `}
+      >
+        <div
+          css={css`
+            display: flex;
+            align-items: center;
+            gap: 16px;
+            flex: 1 0 0;
+          `}
+        >
+          <img src={TranscriptImage} height="28" width="28" alt="transcript" />{" "}
+          <h1
+            css={css`
+              color: var(--leo-color-text-primary);
+              font-family: var(--Family-Headings, Poppins);
+              font-size: var(--Size-H2, 28px);
+              font-style: normal;
+              font-weight: 600;
+              line-height: var(--Line-height-H2, 36px);
+              letter-spacing: var(--Letter-spacing-Headings, -0.5px);
+              text-wrap: nowrap;
+            `}
+          >
+            Meeting Transcript
+          </h1>
+        </div>
+        <Button
+          css={css`
+            display: flex;
+            min-height: 44px;
+            padding: var(--leo-spacing-l) var(--leo-spacing-xl);
+            justify-content: center;
+            align-items: center;
+            max-width: fit-content;
+          `}
+          kind="outline"
+          onClick={() => {
+            const link = document.createElement("a");
+            link.href = transcript.blobUrl;
+            link.download = `${transcript.id}.txt`;
+            link.click();
```

Comment:
any pitfalls with this approach? we could just add a `Content-Disposition` header in the transcript response

Reply:
I am not aware of any issues with this. I believe this is the standard way to force a download of a file from javascript. The alternative is we need to make a change on the server-side with a parameter / route which adds the Content-Disposition header.

Reply:
okay. your call! will resolve for now since this seems okay